tdm/docs/architecture.md

```diff
@@ -0,0 +1,262 @@
+
+1. The device detection module. It monitors udev events to detect PCI devices
+   addition and removal from the guest emulated PCI bus. This module calls into
+   the TDM core to notify it about both event types. The TDM core then relies on
+   the following two modules to verify, attest and accept newly detected
+   devices. Only TDM-accepted TDIs are visible and available to the rest of the
+   guest software stack.
+2. The ABI plugins module. This is an abstraction over the Linux CoCo
+   vendor-specific ABI and the TDM core calls into this module in order to
+   communicate with the TSM. The ABI plugins module abstracts the
+   vendor-specific TSM interfaces in order for the TDM to:
+   * Retrieve a certificate chain for any newly detected device.
+   * Retrieve an attestation evidence for any newly detected device. This is
+     optional as the evidence may be delivered as part of the device certificate
+     chain.
+   * Check the physical device link state and verify that it is confidentiality
+     and integrity protected, as provided by the PCI IDE protocol. The TSM
+     manages the IDE keys and must track the device physical link state for any
+     assigned devices.
+   * Check for the TDI state in order to verify that it is in the (TDISP) locked
+     state.
+   * Check that all emulated MMIO ranges are correctly (in respect to their size
+     and order in the guest address space) mapped.
+   * Accept or reject an assigned TDI.
+3. The device attestation module. Once the TDM has authenticated and verified an
+   assigned TDI, it can attest it through local or remote attestation before
+   finally accepting it. The TDM core fetches an attestation evidence through
+   the ABI plugin module and forwards it to the device attestation module. This
+   module then verifies the device evidence either locally or remotely, depending
+   on a guest-specific TDM policy, and returns an attestation results back to the
+   TDM core.
+
+### Device Detection
+
+### TEE ABI Plugins
+
+
+
+#### Plugin Interface
+
+A Rust trait called by the TDM core to interact with the TSM.
+(Not needed if Linux provides a converged TEE userspace API.
+
+``` rust
+
+
+
+```
+
+### Device Attestation
+
+## Theory of Operations
+
+### TEE-IO Acceptance Flow
+
+#### Detection
+
+#### TDI Security Attributes Check
+
+#### Device Attestation
+
+##### Local Attestation
+
+TDM is a Relying Party.
+TDM to use part of the Attestation Service crates
+
+##### Remote Attestation
+
+## Glossary
+
+| Term | Acronym | Definition |
+|:-----|:-------:|------------|
+| Confidential Computing                 | CoCo   | The protection of data in use by performing computation in a Hardware-based TEE.
+| Confidential VM                        | CVM    | A confidential computing virtual machine. Same as a TVM.
+| Device Interface                       | DI     | TDISP term, Device Interface. Same as a TDI. A DI can be a Virtual Function (VF) or a Physical Function (PF).
```

Comment:
What is the difference between "DI" and "TDI"?
If they are same, why we need to define "DI" ?

Reply:
Removed, thanks.